### ISSUE TITLE
Allow creating wildcard domains in `baseDomains`

### ISSUE THREAD

**Issue by szlend** [Janik-Haag/NixOS-DNS#6]
Some services need wildcard dns support so it makes sense to configure them in `baseDomains`. Wildcard domains work fine in  `dns.Config.extraConfig.zones.<name>`, but not in `baseDomains` because it tries to create a zone file called `*.example.com`. I tested a fix with a local checkout of NixOS-DNS where I replace the `*` with `_wildcard` and that fixed the issue. But I'm not sure if that's the preferred solution.

**Comment by Janik-Haag** [comment]
if you have
```nix
  networking.domains.baseDomains = {
    "example.com" = {};
    "*.example.com" = {};
  };
  networking.domains.subDomains = {
    "*.example.com" = {
      # your records
    };
  };
```
it should only create a zonefile for example.com and merge the values of `*.example.com`, I think this behavior might be undocumented :see_no_evil: 
Currently basedomains are not meant to be rendered in the output and changing that would probably create some nasty edgecases. I think you should be able to do what you want with the example above, also I have to admit that the semantics are a bit ugly. Depending on what you want to do dname records might also be a solution.

Hope this helps ^^

**Comment by szlend** [comment]
Thanks for the explanation! I did actually try this initially. Turns out I had to remove the root domain from my `dns.nix` zones and then it worked like expected.

I also tried 
```nix
networking.domains.baseDomains = {
  "example.com" = { ... };
};
networking.domains.subDomains = {
  "example.com" = { };
  "*.example.com" = { a.data = "1.2.3.4" };
};
```

And this completely ignored the records from  `*.example.com` and just inherited them from `example.com`.

The interaction between `dnsConfig.extraConfig`, `baseDomains`, `subDomains` are very non-obvious imo.
